ros_buildfarm/templates/devel/devel_job.xml.em

```diff
@@ -191,7 +191,7 @@ if pull_request:
         ' -e=TRAVIS=$TRAVIS' +
         ' -v $WORKSPACE/ros_buildfarm:/tmp/ros_buildfarm:ro' +
         ' -v $WORKSPACE/catkin_workspace:/tmp/catkin_workspace' +
-        ' -v ~/.ccache:/home/buildfarm/.ccache' +
+        ' -v $CCACHE_DIR:/home/buildfarm/.ccache' +
```

Comment:
> ... but I don't see where it is being used ...

@dirk-thomas Originally, the variable `CCACHE_DIR` was used here to mount the user's ccache folder into the docker container. However, your comment in https://github.com/ros-infrastructure/ros_buildfarm/pull/582#issuecomment-432342615 made me revert these changes. 
But, when I grep for `~/.ccache:` only those *_job.xml.em files are listed. Hence, they should get this change!
```
ros_buildfarm/templates/release/binarydeb_job.xml.em:        ' -v ~/.ccache:/home/buildfarm/.ccache' + \
ros_buildfarm/templates/release/binarydeb_job.xml.em:        ' -v ~/.ccache:/home/buildfarm/.ccache' +
ros_buildfarm/templates/devel/devel_job.xml.em:        ' -v ~/.ccache:/home/buildfarm/.ccache' +
ros_buildfarm/templates/devel/devel_job.xml.em:        ' -v ~/.ccache:/home/buildfarm/.ccache' +
```

Reply:
In the first version the environment variable was only set in local scripts but not for buildfarm jobs but unconditionally used to mount that directory. Hence my first comments: https://github.com/ros-infrastructure/ros_buildfarm/pull/582#issuecomment-432319236 and https://github.com/ros-infrastructure/ros_buildfarm/pull/582#issuecomment-432342615

Then you removed the usage during the mount which leaves just the environment variable without any effect: hence https://github.com/ros-infrastructure/ros_buildfarm/pull/582#issuecomment-432430274 and https://github.com/ros-infrastructure/ros_buildfarm/pull/582#issuecomment-433968179

The patch needs to be updated to set the environment variable (when run locally as well as on the buildfarm) and then it can be used in the mount parameter.

Reply:
OK, I reverted the revert. If there is anything more missing, I think you are by far the better expert on this. I just noticed, that the `CCACHE_DIR` is not considered for mounting the appropriate folder into the docker container. I guess, all the bits and pieces are there and I'am sure you will quickly finalize this.

Reply:
The patch is now back in the original state. While that works when you run the build locally on Jenkins the environment variable won't be defined yet and therefore the mount argument `-v $CCACHE_DIR:/home/buildfarm/.ccache` will expand to `-v :/home/buildfarm/.ccache` and therefore fail.

The environment variable needs to be not only defined in the local scripts but also in the case of running within a Jenkins job.

Reply:
I agree. Please add the missing definition in the corresponding file. I don't know about the details of the build farm scripts to do this.